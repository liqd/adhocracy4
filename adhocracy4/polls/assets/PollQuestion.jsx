import React, { useEffect, useState } from 'react'
import django from 'django'

export const PollQuestion = (props) => {
  const getUserAnswer = () => {
    const userAnswerId = props.question.other_choice_user_answer
    const userAnswer = props.question.other_choice_answers.find(oc => oc.vote_id === userAnswerId)
    return (userAnswerId && userAnswer)
      ? userAnswer.answer
      : ''
  }

  const multiHelpText = props.question.multiple_choice ? <div className="poll__help-text">{django.gettext('Multiple answers are possible.')}</div> : null
  const questionHelpText = props.question.help_text ? <div className="poll__help-text">{props.question.help_text}</div> : null
  const [userChoices, setUserChoices] = useState(props.question.userChoices)
  const [otherChoiceAnswer, setOtherChoiceAnswer] = useState(getUserAnswer())

  const handleSingleChange = (event) => {
    const choiceId = parseInt(event.target.value)
    setUserChoices([choiceId])
    props.onSingleChange(props.question.id, choiceId)
  }

  const handleMultiChange = (event) => {
    const choiceId = parseInt(event.target.value)
    setUserChoices(prevState => [...prevState, choiceId])
    props.onMultiChange(props.question.id, choiceId)
  }

  const handleOtherChange = (event) => {
    const otherAnswer = event.target.value
    setOtherChoiceAnswer(otherAnswer)
    props.onOtherChange(props.question.id, otherAnswer)
  }

  useEffect(() => setUserChoices(props.question.userChoices))

  return (
    <form>
      <div className="poll">
        <h2>{props.question.label}</h2>
        {questionHelpText}
        {multiHelpText}
        <div className="poll__rows">
          {
            props.question.choices.map((choice, i) => {
              const checked = userChoices.indexOf(choice.id) !== -1

              if (!props.question.multiple_choice) {
                return (
                  <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-single'}>
                    <input
                      className="poll-row__radio radio__input"
                      type="radio"
                      name="question"
                      id={'id_choice-' + choice.id + '-single'}
                      value={choice.id}
                      checked={checked}
                      onChange={(event) => { handleSingleChange(event) }}
                      disabled={!props.question.authenticated}
                    />
                    <span className="radio__text">{choice.label}</span>
                    {choice.is_other_choice &&
                      <input
                        className="input-group__input"
                        type="text"
                        name="question"
                        value={otherChoiceAnswer}
                        id={'id_choice-' + choice.id + '-single-answer'}
                        onChange={(event) => { handleOtherChange(event) }}
                        disabled={!props.question.authenticated}
                      />}
                  </label>
                )
              } else {
                return (
                  <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-multiple'}>
                    <input
                      className="poll-row__radio radio__input"
                      type="checkbox"
                      name="question"
                      id={'id_choice-' + choice.id + '-multiple'}
                      value={choice.id}
                      checked={checked}
                      onChange={(event) => { handleMultiChange(event) }}
                      disabled={!props.question.authenticated}
                    />
                    <span className="radio__text radio__text--checkbox">{choice.label}</span>
                    {choice.is_other_choice &&
                      <input
                        className="input-group__input"
                        type="text"
                        name="question"
                        id={'id_choice-' + choice.id + '-single'}
                        value={otherChoiceAnswer}
                        onChange={(event) => { handleOtherChange(event) }}
                        disabled={!props.question.authenticated}
                      />}
                  </label>
                )
              }
            })
          }
        </div>
      </div>
    </form>
  )
}
