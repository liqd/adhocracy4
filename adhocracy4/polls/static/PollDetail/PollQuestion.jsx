import React, { useEffect, useState } from 'react'
import django from 'django'
import { CharCounter } from './CharCounter'
import ErrorList from '../../../static/ErrorList'

const translated = {
  multiple: django.gettext('Multiple answers are possible.'),
  other: django.gettext('other')
}

export const PollQuestion = (props) => {
  // | Function to define state

  const getUserAnswer = () => {
    const userAnswerId = props.question.other_choice_user_answer
    const userAnswer = props.question.other_choice_answers.find(oc => oc.vote_id === userAnswerId)
    return props.question.other_choice_answer
      ? props.question.other_choice_answer
      : ((userAnswerId && userAnswer)
          ? userAnswer.answer
          : ''
        )
  }

  const [userChoices, setUserChoices] = useState(props.question.userChoices)
  const [otherChoiceAnswer, setOtherChoiceAnswer] = useState(getUserAnswer())
  const [errors, setErrors] = useState()
  const multiHelpText = props.question.multiple_choice ? <div className="poll__help-text">{translated.multiple}</div> : null
  const questionHelpText = props.question.help_text ? <div className="poll__help-text">{props.question.help_text}</div> : null
  const maxlength = 250

  useEffect(() => {
    setUserChoices(props.question.userChoices)
    setErrors(props.errors)
  }, [props.question.userChoices, props.errors])

  const handleSingleChange = (event, isOther) => {
    const choiceId = parseInt(event.target.value)
    setUserChoices([choiceId])
    props.onSingleChange(props.question.id, choiceId)
    if (!isOther) {
      setOtherChoiceAnswer('')
      props.onOtherChange(props.question.id, '', findOtherChoice())
    }
  }

  const handleMultiChange = (event, isOther) => {
    const choiceId = parseInt(event.target.value)
    setUserChoices(prevState => [...prevState, choiceId])
    props.onMultiChange(props.question.id, choiceId)
    if (!isOther) {
      setOtherChoiceAnswer('')
      props.onOtherChange(props.question.id, '', findOtherChoice())
    }
  }

  const handleOtherChange = (event) => {
    const otherAnswer = event.target.value
    setOtherChoiceAnswer(otherAnswer)
    props.onOtherChange(props.question.id, otherAnswer)
  }

  const findOtherChoice = () => {
    return props.question.choices.find(c => c.is_other_choice)
  }

  return (
    <form>
      <div className="poll">
        <h2>{props.question.label}</h2>
        {questionHelpText}
        {multiHelpText}
        <div className="poll__rows">
          {
            props.question.choices.map((choice, i) => {
              const checked = userChoices.indexOf(choice.id) !== -1

              if (!props.question.multiple_choice) {
                return (
                  <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-single'}>
                    <input
                      className="poll-row__radio"
                      type="radio"
                      name="question"
                      id={'id_choice-' + choice.id + '-single'}
                      value={choice.id}
                      checked={checked}
                      onChange={(event) => { handleSingleChange(event, choice.is_other_choice) }}
                      disabled={!props.question.authenticated || props.question.isReadOnly}
                    />
                    <span className="radio__text">{choice.is_other_choice ? translated.other : choice.label}</span>
                    {choice.is_other_choice &&
                      <>
                        <input
                          className="form-control"
                          type="text"
                          name="question"
                          value={otherChoiceAnswer}
                          id={'id_choice-' + choice.id + '-other'}
                          onChange={(event) => { handleOtherChange(event) }}
                          disabled={!props.question.authenticated || props.question.isReadOnly || !checked}
                          maxLength={maxlength}
                        />
                        {checked
                          ? (
                            <>
                              <div className="poll__char-counter">
                                <CharCounter value={otherChoiceAnswer} max={maxlength} />
                              </div>
                              <ErrorList errors={errors} field={choice.id} />
                            </>
                            )
                          : null}
                      </>}
                  </label>
                )
              } else {
                return (
                  <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-multiple'}>
                    <input
                      className="poll-row__radio"
                      type="checkbox"
                      name="question"
                      id={'id_choice-' + choice.id + '-multiple'}
                      value={choice.id}
                      checked={checked}
                      onChange={(event) => { handleMultiChange(event, choice.is_other_choice) }}
                      disabled={!props.question.authenticated || props.question.isReadOnly}
                    />
                    <span className="radio__text radio__text--checkbox">{choice.is_other_choice ? translated.other : choice.label}</span>
                    {choice.is_other_choice &&
                      <>
                        <input
                          className="form-control"
                          type="text"
                          name="question"
                          id={'id_choice-' + choice.id + '-other'}
                          value={otherChoiceAnswer}
                          onChange={(event) => { handleOtherChange(event) }}
                          disabled={!props.question.authenticated || props.question.isReadOnly || !checked}
                          maxLength={maxlength}
                        />
                        {checked
                          ? (
                            <>
                              <div className="poll__char-counter">
                                <CharCounter value={otherChoiceAnswer} max={maxlength} />
                              </div>
                              <ErrorList errors={errors} field={choice.id} />
                            </>
                            )
                          : null}
                      </>}
                  </label>
                )
              }
            })
          }
        </div>
      </div>
    </form>
  )
}
