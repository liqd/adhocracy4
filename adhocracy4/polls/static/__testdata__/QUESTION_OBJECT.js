export const QUESTION_OBJECT = {
  id: 1,
  label: 'My Question',
  help_text: '',
  multiple_choice: false,
  is_open: false,
  isReadOnly: false,
  authenticated: true,
  choices: [
    {
      id: 1,
      label: 'cool1',
      count: 0,
      is_other_choice: false
    },
    {
      id: 2,
      label: 'yeah',
      count: 0,
      is_other_choice: false
    }
  ],
  userChoices: [],
  answers: [],
  userAnswer: '',
  other_choice_answers: [],
  other_choice_user_answer: '',
  totalVoteCount: 0,
  totalVoteCountMulti: 0,
  totalAnswerCount: 0
}
