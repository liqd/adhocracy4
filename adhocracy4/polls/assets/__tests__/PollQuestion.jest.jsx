// tools needed for testing
import React from 'react'
import { render, fireEvent } from '@testing-library/react'

// component and related data to be tested
import { PollQuestion } from 'adhocracy4/polls/assets/PollQuestion.jsx'
import { QUESTION_OBJECT } from 'adhocracy4/polls/assets/__tests__/__testdata__/QUESTION_OBJECT'

describe('render <PollQuestion> with...', () => {
  test('-> single-choice -> non-open-asnwers', () => {
    const tree = render(<PollQuestion question={QUESTION_OBJECT} />)
    expect(tree).toMatchSnapshot()
  })

  test('-> single-choice -> open-asnwers', () => {
    const singleOpenQuestion = { ...QUESTION_OBJECT }
    singleOpenQuestion.is_open = true
    const tree = render(<PollQuestion question={singleOpenQuestion} />)
    expect(tree).toMatchSnapshot()
  })

  test('-> multiple-choice -> non-open-asnwers', () => {
    const multiQuestion = { ...QUESTION_OBJECT }
    multiQuestion.multiple_choice = true
    const tree = render(<PollQuestion question={multiQuestion} />)
    expect(tree).toMatchSnapshot()
  })

  test('-> multiple-choice -> open-asnwers', () => {
    const multiOpenQuestion = { ...QUESTION_OBJECT }
    multiOpenQuestion.multiple_choice = true
    multiOpenQuestion.is_open = true
    const tree = render(<PollQuestion question={multiOpenQuestion} />)
    expect(tree).toMatchSnapshot()
  })
})

describe('calling prop-passed functions...', () => {
  const changedFn = jest.fn()
  const otherChangedFn = jest.fn()
  const multiChangedFn = jest.fn()

  test('call onSingleChange', () => {
    const tree = render(
      <PollQuestion
        question={QUESTION_OBJECT}
        onSingleChange={changedFn}
        onOtherChange={otherChangedFn}
      />
    )
    const choiceRadio1 = tree.container.querySelector('#id_choice-1-single')
    expect(choiceRadio1.checked).toBe(false)
    fireEvent.click(choiceRadio1)
    expect(changedFn).toHaveBeenCalled()
    expect(otherChangedFn).toHaveBeenCalled()
  })

  test('call onSingleChange -> is_other_choice', () => {
    const singleOpenQuestion = { ...QUESTION_OBJECT }
    singleOpenQuestion.choices[0].is_other_choice = true
    const tree = render(
      <PollQuestion
        question={singleOpenQuestion}
        onSingleChange={changedFn}
        onOtherChange={otherChangedFn}
      />
    )
    const choiceRadio = tree.container.querySelector('#id_choice-1-other')
    const choiceTextInput = tree.container.querySelector('#id_choice-1-other')
    expect(choiceRadio.checked).toBe(false)
    expect(choiceTextInput.value).toBe('')
    fireEvent.click(choiceRadio)
    fireEvent.change(choiceTextInput, { target: { value: 'something' } })
    expect(otherChangedFn).toHaveBeenCalled()
    expect(otherChangedFn).toHaveBeenCalledWith(1, 'something')
  })

  test('call onMultiChange', () => {
    const multiQuestion = { ...QUESTION_OBJECT }
    multiQuestion.multiple_choice = true
    multiQuestion.choices[0].is_other_choice = false
    const tree = render(
      <PollQuestion
        question={multiQuestion}
        onMultiChange={multiChangedFn}
        onOtherChange={otherChangedFn}
      />
    )
    const choiceCheckbox1 = tree.container.querySelector('#id_choice-1-multiple')
    expect(choiceCheckbox1.checked).toBe(false)
    fireEvent.click(choiceCheckbox1)
    expect(multiChangedFn).toHaveBeenCalled()
    expect(otherChangedFn).toHaveBeenCalled()
  })

  test('call onMultiChange -> is_other_choice', () => {
    const multiOpenQuestion = { ...QUESTION_OBJECT }
    multiOpenQuestion.multiple_choice = true
    multiOpenQuestion.choices[0].is_other_choice = true
    const tree = render(
      <PollQuestion
        question={multiOpenQuestion}
        onMultiChange={multiChangedFn}
        onOtherChange={otherChangedFn}
      />
    )
    const choiceCheckbox = tree.container.querySelector('#id_choice-1-multiple')
    const choiceTextInput = tree.container.querySelector('#id_choice-1-other')
    expect(choiceCheckbox.checked).toBe(false)
    expect(choiceTextInput.value).toBe('')
    fireEvent.click(choiceCheckbox)
    fireEvent.change(choiceTextInput, { target: { value: 'something' } })
    expect(otherChangedFn).toHaveBeenCalled()
    expect(otherChangedFn).toHaveBeenCalledWith(1, 'something')
  })
})

test('initialize with function getUserAnswer -> with user answer', () => {
  const userAnswerQuestion = { ...QUESTION_OBJECT }
  userAnswerQuestion.other_choice_answers = [{ answer: 'antwoord', vote_id: 1 }]
  userAnswerQuestion.other_choice_user_answer = 1

  const tree = render(
    <PollQuestion
      question={userAnswerQuestion}
    />
  )
  const choiceTextInput = tree.container.querySelector('#id_choice-1-other')
  expect(choiceTextInput.value).toBe('antwoord')
})
