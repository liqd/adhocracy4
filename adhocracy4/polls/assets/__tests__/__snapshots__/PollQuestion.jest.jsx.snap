// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <PollQuestion> with... -> multiple-choice -> non-open-asnwers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="poll"
        >
          <h2>
            My Question
          </h2>
          <div
            class="poll__help-text"
          >
            mock text
          </div>
          <div
            class="poll__rows"
          >
            <label
              class="poll-row radio"
              for="id_choice-1-multiple"
            >
              <input
                class="poll-row__radio"
                id="id_choice-1-multiple"
                name="question"
                type="checkbox"
                value="1"
              />
              <span
                class="radio__text radio__text--checkbox"
              >
                cool1
              </span>
            </label>
            <label
              class="poll-row radio"
              for="id_choice-2-multiple"
            >
              <input
                class="poll-row__radio"
                id="id_choice-2-multiple"
                name="question"
                type="checkbox"
                value="2"
              />
              <span
                class="radio__text radio__text--checkbox"
              >
                yeah
              </span>
            </label>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="poll"
      >
        <h2>
          My Question
        </h2>
        <div
          class="poll__help-text"
        >
          mock text
        </div>
        <div
          class="poll__rows"
        >
          <label
            class="poll-row radio"
            for="id_choice-1-multiple"
          >
            <input
              class="poll-row__radio"
              id="id_choice-1-multiple"
              name="question"
              type="checkbox"
              value="1"
            />
            <span
              class="radio__text radio__text--checkbox"
            >
              cool1
            </span>
          </label>
          <label
            class="poll-row radio"
            for="id_choice-2-multiple"
          >
            <input
              class="poll-row__radio"
              id="id_choice-2-multiple"
              name="question"
              type="checkbox"
              value="2"
            />
            <span
              class="radio__text radio__text--checkbox"
            >
              yeah
            </span>
          </label>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render <PollQuestion> with... -> multiple-choice -> open-asnwers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="poll"
        >
          <h2>
            My Question
          </h2>
          <div
            class="poll__help-text"
          >
            mock text
          </div>
          <div
            class="poll__rows"
          >
            <label
              class="poll-row radio"
              for="id_choice-1-multiple"
            >
              <input
                class="poll-row__radio"
                id="id_choice-1-multiple"
                name="question"
                type="checkbox"
                value="1"
              />
              <span
                class="radio__text radio__text--checkbox"
              >
                cool1
              </span>
            </label>
            <label
              class="poll-row radio"
              for="id_choice-2-multiple"
            >
              <input
                class="poll-row__radio"
                id="id_choice-2-multiple"
                name="question"
                type="checkbox"
                value="2"
              />
              <span
                class="radio__text radio__text--checkbox"
              >
                yeah
              </span>
            </label>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="poll"
      >
        <h2>
          My Question
        </h2>
        <div
          class="poll__help-text"
        >
          mock text
        </div>
        <div
          class="poll__rows"
        >
          <label
            class="poll-row radio"
            for="id_choice-1-multiple"
          >
            <input
              class="poll-row__radio"
              id="id_choice-1-multiple"
              name="question"
              type="checkbox"
              value="1"
            />
            <span
              class="radio__text radio__text--checkbox"
            >
              cool1
            </span>
          </label>
          <label
            class="poll-row radio"
            for="id_choice-2-multiple"
          >
            <input
              class="poll-row__radio"
              id="id_choice-2-multiple"
              name="question"
              type="checkbox"
              value="2"
            />
            <span
              class="radio__text radio__text--checkbox"
            >
              yeah
            </span>
          </label>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render <PollQuestion> with... -> single-choice -> non-open-asnwers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="poll"
        >
          <h2>
            My Question
          </h2>
          <div
            class="poll__rows"
          >
            <label
              class="poll-row radio"
              for="id_choice-1-single"
            >
              <input
                class="poll-row__radio"
                id="id_choice-1-single"
                name="question"
                type="radio"
                value="1"
              />
              <span
                class="radio__text"
              >
                cool1
              </span>
            </label>
            <label
              class="poll-row radio"
              for="id_choice-2-single"
            >
              <input
                class="poll-row__radio"
                id="id_choice-2-single"
                name="question"
                type="radio"
                value="2"
              />
              <span
                class="radio__text"
              >
                yeah
              </span>
            </label>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="poll"
      >
        <h2>
          My Question
        </h2>
        <div
          class="poll__rows"
        >
          <label
            class="poll-row radio"
            for="id_choice-1-single"
          >
            <input
              class="poll-row__radio"
              id="id_choice-1-single"
              name="question"
              type="radio"
              value="1"
            />
            <span
              class="radio__text"
            >
              cool1
            </span>
          </label>
          <label
            class="poll-row radio"
            for="id_choice-2-single"
          >
            <input
              class="poll-row__radio"
              id="id_choice-2-single"
              name="question"
              type="radio"
              value="2"
            />
            <span
              class="radio__text"
            >
              yeah
            </span>
          </label>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render <PollQuestion> with... -> single-choice -> open-asnwers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="poll"
        >
          <h2>
            My Question
          </h2>
          <div
            class="poll__rows"
          >
            <label
              class="poll-row radio"
              for="id_choice-1-single"
            >
              <input
                class="poll-row__radio"
                id="id_choice-1-single"
                name="question"
                type="radio"
                value="1"
              />
              <span
                class="radio__text"
              >
                cool1
              </span>
            </label>
            <label
              class="poll-row radio"
              for="id_choice-2-single"
            >
              <input
                class="poll-row__radio"
                id="id_choice-2-single"
                name="question"
                type="radio"
                value="2"
              />
              <span
                class="radio__text"
              >
                yeah
              </span>
            </label>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="poll"
      >
        <h2>
          My Question
        </h2>
        <div
          class="poll__rows"
        >
          <label
            class="poll-row radio"
            for="id_choice-1-single"
          >
            <input
              class="poll-row__radio"
              id="id_choice-1-single"
              name="question"
              type="radio"
              value="1"
            />
            <span
              class="radio__text"
            >
              cool1
            </span>
          </label>
          <label
            class="poll-row radio"
            for="id_choice-2-single"
          >
            <input
              class="poll-row__radio"
              id="id_choice-2-single"
              name="question"
              type="radio"
              value="2"
            />
            <span
              class="radio__text"
            >
              yeah
            </span>
          </label>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
